### PR TITLE
Remove all build-time dependencies on mono

### PR DESCRIPTION
**Pull request by mellinoe**
None of these checks are necessary any longer, as we have moved to using CoreCLR for all of our .NET-related build tools.

The only remaining use of anything mono-related is the PCL reference assemblies, which are not required to build. If the PCL reference assemblies are not found, then (harmless) build warnings will be emitted. Does anyone know of a way to suppress the warning messages regarding missing reference assemblies? This is because we are claiming our libraries are targeting .NETPortable, Profile7.

@weshaggard , @joperezr , @asiekierka

### CODE REVIEW

File: override.targets [hunk 1]
```diff
@@ -2,6 +2,4 @@
   <!--
     Overrides for all other targets (including build tools) can go in this file.
   -->
-
```
Comment (weshaggard) [inline override.targets]: I think we can completely remove the override.targets file as well.

**Comment by weshaggard** [comment]
LGTM thanks for cleaning this up @mellinoe 

cc @ellismg as FYI

**Comment by mellinoe** [comment]
I think there are still some issues when building without the reference assemblies. I believe System.Composition.* relies on them still (The csproj's are overriding the TargetFrameworkProfile property, which causes us to resolve against that profile). Also, in trying to get rid of the warnings if you don't have the reference assemblies, I'm hitting some other issues in various projects which I'm still trying to figure out.

The changes in this PR should be good to go, though, I'll just need to make some more changes (possibly in buildtools) to get rid of the reliance on the PCL reference assemblies.

**Comment by weshaggard** [comment]
Now that you mention that I do remember some libraries like Immutable are using the PCL reference assemblies https://github.com/dotnet/corefx/blob/master/src/System.Collections.Immutable/src/System.Collections.Immutable.csproj#L11 However I think we have everything we need to eliminate that dependency now. It only existed originally because we wanted to make sure it was compiling against the older versions of the contract assemblies for compat reasons but we now have packages with all the older versions so we should be able to match the assembly versions.

**Comment by mellinoe** [comment]
@weshaggard Yeah, that's right. I'm also seeing another, more subtle issue. Some libraries, like System.Security.Cryptography.Cng, have some unused-using-statements (for example, [System.Diagnostics.Contracts](https://github.com/dotnet/corefx/blob/master/src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.Create.cs#L7)) in some files for namespaces that aren't contained in any of the referenced assemblies. For some reason, this doesn't create an error when the PCL reference assemblies are present, but does otherwise. There were a bunch of these issues that I saw when I did a full build. We'll have to go through and remove those using statements before we can do a full build without the PCL reference assemblies.

**Comment by joperezr** [comment]
LGTM. Also there is an issue for this #5743 so I can probably assign you since you did the work and so you can close it. Thanks for this @mellinoe

**Comment by ellismg** [comment]
FYI, @mellinoe When trying in the past to remove all this crap I hit the same warnings due to the lack of reference assemblies and then after a while a bunch of errors.  It may make sense to keep that part for now but clean up the overrides stuff, downloading nuget, etc.  You're call on how much time you have tracking this stuff down.

**Comment by jasonwilliams200OK** [comment]
:+1: 
Hopefully the rabbit hole doesn't go much deeper. Would be really a great deal if we get to say adios to mono related external dependencies with this PR.

**Comment by mellinoe** [comment]
> FYI, @mellinoe When trying in the past to remove all this crap I hit the same warnings due to the lack of reference assemblies and then after a while a bunch of errors

@ericstj helped me figure out some MSBuild properties to set to ignore / suppress / work-around those warnings (at least the ones I was seeing), basically the ones in this commit: https://github.com/mellinoe/corefx/commit/f7a8db8e6e56d6b5ce3551783f66333b601bca81#diff-0b192804a6349e8c26d2b027afbd89a2R436.

After that point, I hit the issues that require code changes, i.e. System.Collections.Immutable using PCL references, and unused-usings causing build errors. I am going to do those separately because it's probably going to touch a lot of random files.

**Comment by mellinoe** [comment]
@dotnet-bot Test Innerloop OSX Release Build and Test

**Comment by mellinoe** [comment]
@mmitche I don't think that the CI is responding to when I request a re-run. Am I saying the right commands?

**Comment by mmitche** [comment]
@mellinoe You are...not sure why that didn't retrigger but something is up right now.  The server is definitely running but the web server appears to not be responding.

**Comment by mmitche** [comment]
@dotnet-bot test Innerloop OSX Release Build and Test

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
Test Innerloop OSX Release Build and Test please
Test Innerloop Ubuntu Release Build and Test please
Test Innerloop Ubuntu Debug Build and Test please
Test Innerloop CentOS7.1 Release Build and Test please

**Comment by manu-silicon** [comment]
Does this mean that the instructions can be updated to remove all references to mono?

**Comment by mellinoe** [comment]
> Does this mean that the instructions can be updated to remove all references to mono?

Soon, hopefully. Right now we still require the PCL reference assemblies to be installed for a few projects, but I am trying to remove that dependency as well. The docs can probably be updated to indicate that mono isn't used for building anything directly. I was planning on updating the docs once I finished removing the PCL reference assembly dependency.

**Pull request merged**